python/pyarrow/tests/test_table.py

```diff
@@ -878,6 +878,55 @@ def test_recordbatch_from_struct_array():
     ))
 
 
+def test_recordbatch_to_struct_array():
+    batch = pa.RecordBatch.from_arrays(
+        [
+            pa.array([1, None], type=pa.int32()),
+            pa.array([None, 1.0], type=pa.float32()),
+        ], ["ints", "floats"]
+    )
+    result = batch.to_struct_array()
+    assert result.equals(pa.array(
+        [{"ints": 1}, {"floats": 1.0}],
+        type=pa.struct([("ints", pa.int32()), ("floats", pa.float32())]),
+    ))
+
+
+def test_table_from_struct_array_invalid():
+    with pytest.raises(TypeError):
```

Comment:
Just a nit: could we add a message check here also? It helps reviewing and also later when looking at the tests, like here: https://github.com/apache/arrow/blob/79c72a67f90719fb1af315a9b2b8caa25d667775/python/pyarrow/tests/test_table.py#L800-L802

Reply:
https://github.com/apache/arrow/pull/38520/commits/d260167fb373abcdb9462e7b03c790538f038dc6